src/main/java/io/codelex/flightplanner/service/FlightService.java

```diff
@@ -40,41 +51,33 @@ public Flight addFlight(AddFlightRequest flightRequest) {
         if (!isArrivalAndDepartureTimeValid(flightRequest)) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
         }
-        if (flightExists(flightRequest)) {
-            throw new ResponseStatusException(HttpStatus.CONFLICT);
-        }
-        airportService.add(airportFrom);
-        airportService.add(airportTo);
         Flight flight = Flight.builder()
                               .from(airportFrom)
                               .to(airportTo)
                               .carrier(flightRequest.getCarrier())
                               .departureTime(flightRequest.getDepartureTime())
                               .arrivalTime(flightRequest.getArrivalTime())
                               .build();
-        return flightRepository.addFlight(flight);
+        if (flightRepository.exists(Example.of(flight))) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+        airportService.add(airportFrom);
```

Comment:
šis tāds nedaudz meh laikam, ka no request paņem objektu un pa taisno liec datubāzē, labāk būtu pašam viņu uztaisīt